deploy.py

```diff
@@ -535,19 +602,55 @@ def main():
     while args.region == 'list':
       region_list = get_gcp_regions(credentials, project_id)
       print('\n'.join(region_list))
-      args.region = (
-          input(
-              'Which region is the GA export in (list for a list of regions)? '
-          ).strip())
+      args.region = (input(
+          'Which region is the GA export in (list for a list of regions)? ').
+                     strip())
+
   if not args.ga_property:
-    args.ga_property = (
-        input(
-            'Please enter the GA property ID you are collecting CWV data with: '
-        ).strip())
+    args.ga_property = (input(
+        'Please enter the GA property ID you are collecting CWV data with: ').
+                        strip())
     if not args.ga_property.isdigit():
       raise SystemExit('Only GA4 properties are supported at this time.')
 
-  deploy_scheduled_materialize_query(project_id, args.region, args.ga_property)
+  # the options are a service account email is provided with the default
+  # credentials, the word default is provided in place of an email address, or
+  # the service_account_email field isn't present at all on the credentials.
+  if not args.iam_service_account:
+    input_msg = 'Please enter the email of the service account to use: '
+    if hasattr(credentials, 'service_account_email'):
+      if credentials.service_account_email == 'default':
+        args.iam_service_account = get_default_service_account_email(
+            project_id, credentials)
+      else:
+        args.iam_service_account = credentials.service_account_email
+
+      input_msg = (
+          'Please note: using the default service account, '
+          f'{args.iam_service_account}, will result in a new role being '
+          'created to allow for the creation and execution of BigQuery '
+          'scheduled queries.\n' + input_msg)
+
+    else:
+      input_msg = ('Please note: your default credentials do not provide a '
+                   'service account. You must provide one here.\n' + input_msg)
+
+    user_service_account = input(input_msg).strip()
+
+    if user_service_account:
+      args.iam_service_account = user_service_account
+
+    if args.iam_service_account:
+      add_roles_to_service_account(args.iam_service_account, project_id,
+                                   credentials)
+    else:
+      raise SystemExit(
+          'You must provide a service account for deploying and '
+          'running the solution to continue. Please create a service account '
+          'and try again.')
```

Comment:
1. I think there is no need for asking user for service account on line 638 of we were able to get it from credentials on line 626 or 623 So I suggest moving this user input request under else block (indicating credentials didn't have any service account)

2. Warning about adding new role applies to all the cases, so I think we better move it before we actually start creating and adding role

@adamread Let me know if you don't agree

```suggestion
  if not args.iam_service_account:
    input_msg = 'Please enter the email of the service account to use: '
    if hasattr(credentials, 'service_account_email'):
      if credentials.service_account_email == 'default':
        args.iam_service_account = get_default_service_account_email(
            project_id, credentials)
      else:
        args.iam_service_account = credentials.service_account_email

    else:
      input_msg = ('Please note: your default credentials do not provide a '
                   'service account. You must provide one here.\n' + input_msg)
      user_service_account = input(input_msg).strip()

    if user_service_account:
      args.iam_service_account = user_service_account

    if args.iam_service_account:
      print(
          'Please note: using service account '
          f'{args.iam_service_account}. It will result in a new role being '
          'created to allow for the creation and execution of BigQuery '
          'scheduled queries.\n')
      add_roles_to_service_account(args.iam_service_account, project_id,
                                   credentials)
    else:
      raise SystemExit(
          'You must provide a service account for deploying and '
          'running the solution to continue. Please create a service account '
          'and try again.')
```

Reply:
Except service accounts logic, looks good to me and works! I will also test email alerts functions shortly and update separately. Thanks @adamread

Reply:
My thought here was about transparency. I wanted the user to know that we were using the service account we found and give them the opportunity to change it. This version doesn't give the user a chance to decide which service account to use. They may decide to stop and create a service account specifically for this app if they know a role will be added to the default service account.

Reply:
Gotcha, thanks for clarification, makes total sense 👍